src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -153,7 +153,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Overrides Function InstrumentReturnStatement(original As BoundReturnStatement, rewritten As BoundStatement) As BoundStatement
             Dim previous As BoundStatement = MyBase.InstrumentReturnStatement(original, rewritten)
             If Not original.IsEndOfMethodReturn Then
-                Return AddDynamicAnalysis(original, previous)
+                If original.ExpressionOpt IsNot Nothing Then
+                    ' Synthesized return statements require instrumentation if they return values,
+                    ' e.g. in simple expression lambdas.
```

Comment:
Do we hit this case with expression lambdas? It sounds like `Not IsEndOfMethodReturn` refers to an `Exit` statement. Are there cases where `ExpressionOpt IsNot Nothing`?